src/query/api/v1/handler/influxdb/rewrite.go

```diff
@@ -0,0 +1,54 @@
+/*
+ * Author: Markus Stenberg <mstenber@aiven.io>
+ *
+ * Copyright (c) 2019 Aiven Oy
+ */
+
+package influxdb
+
+import "regexp"
+
+type regexpRewriter struct {
+	okRe          *regexp.Regexp
+	brokenStartRe *regexp.Regexp
+	brokenRestRe  *regexp.Regexp
+}
+
+func newRegexpRewriter(ok, brokenStart, brokenRest string) *regexpRewriter {
+	return &regexpRewriter{
+		okRe:          regexp.MustCompile(ok),
+		brokenStartRe: regexp.MustCompile(brokenStart),
+		brokenRestRe:  regexp.MustCompile(brokenRest),
+	}
+}
+
+func (self *regexpRewriter) ToValid(input []byte) []byte {
+	matched := self.okRe.Match(input)
+	// Fast path - costs only one regexp evaluation
+	// (This should be the common one)
+	if matched || len(input) == 0 {
+		return input
+	}
+	// Slow path - costs allocations etc
+	input = self.brokenStartRe.ReplaceAll(input, []byte("_"))
+	input = self.brokenRestRe.ReplaceAll(input, []byte("_"))
+	return input
+}
+
+// Stateful utility, which handles both __name__ ('metric') tag, as
```

Comment:
Does this need to be stateful, or can it just be utility methods? Doesn't look like it changes behavior at all

Reply:
That's bit historic comment anyway, guess I'll just remove stateful; internally it has some state (regexps in the old version, and now []byte(256) tables in the new one).